src/components/ArticlesList.jsx

```diff
@@ -2,14 +2,30 @@ import ArticleCard from "./ArticleCard";
 import Skeleton from "@mui/material/Skeleton";
 import { Link } from "react-router-dom";
 import { ArticleOverviewSkeleton } from "../utils/loadingSkeletons";
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+export default function ArticlesList({
+  loading,
+  allArticles,
+  setCurrentTopic,
+}) {
+  const { topic } = useParams();
+
+  useEffect(() => {
+    setCurrentTopic(topic);
+  }, []);
+
+  const filteredArticles = allArticles.filter((article) => {
+    return topic === "all" ? true : article.topic === topic;
+  });
```

Comment:
You need to use the api to filter topics instead of fetching all and locally filtering, please change this.